weave/trait/power/on_off_trait.proto

```diff
@@ -0,0 +1,75 @@
+/*
+ *
+ *    Copyright (c) 2019 Google LLC.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This trait specifies a Weave Common trait that generalizes how
+ *      any Weave device resource can present power on/off functionality.
+ *
+ */
+
+syntax = "proto3";
+
+package weave.trait.power;
+
+import "google/protobuf/wrappers.proto";
+import "wdl/wdl_options.proto";
+import "weave/common/command_response_status.proto";
+
+option objc_class_prefix = "SCM";
+
+option java_outer_classname = "WeaveInternalOnOffTrait";
+
+/**
+ * A binary state to represent on or off.
+ */
+message OnOffTrait {
+  option (wdl.message_type) = TRAIT;
+  option (wdl.trait) = {
+    stability: ALPHA,
+    id: 0x001F,
+    vendor_id: 0x0000,
+    version: 1
+  };
+
+  enum OnOffStatus {
+    ON_OFF_STATUS_UNSPECIFIED = 0;
+    ON_OFF_STATUS_ON = 1;
+    ON_OFF_STATUS_OFF = 2;
+    ON_OFF_STATUS_UNKNOWN = 3;
+  }
+
+  google.protobuf.BoolValue on_off_status = 1 [(wdl.prop) = {nullable: true}];
```

Comment:
When using a boolean, the name of the field has to reflect what 'true' means. In this, it would be 'is_on'.

Reply:
Done